data_steward/utils/bq.py

```diff
@@ -41,14 +145,88 @@ def list_datasets(project_id):
 def get_dataset(project_id, dataset_id):
     """
     Return the dataset object
-    :param project_id: 
-    :param dataset_id: 
+
+    :param project_id: string name of the project id
+    :param dataset_id: string name or Dataset Reference object for the dataset
+
     :return: the dataset object
+
+    :raises: an GoogleAPIError (404) if the dataset is not found
     """
     client = get_client(project_id)
     return client.get_dataset(dataset_id)
 
 
+def create_dataset(project_id,
+                   dataset_id,
+                   description,
+                   labels={},
+                   exists_ok=False):
+    """
+    Create the dataset reference.
+
+    :param project_id:  string name of the project to search for a dataset
+    :param dataset_id:  string name of the dataset id to return a reference of
+
+    :return: a dataset reference object.
+
+    :raises: any GoogleAPIError that is not a 404 error.
+        google.api_core.exceptions.Conflict if the dataset already exists
+    """
+    if description.isspace() or not description:
+        raise RuntimeError("you must provide a description.")
+
+    if not project_id:
+        raise RuntimeError(
+            "You must specify the project_id for the project containing the dataset"
+        )
+
+    if not dataset_id:
+        raise RuntimeError("You must provide a dataset_id")
+
+    client = get_client(project_id)
+
+    dataset_id = "{}.{}".format(client.project, dataset_id)
+
+    # Construct a full Dataset object to send to the API.
+    dataset = bigquery.Dataset(dataset_id)
+
```

Comment:
interesting concept.  I'll see how I can work this in.

Reply:
I added the description and labels to this method.  Thank you for that catch.  I've left the `get_or_create_dataset` definition and changed `create_dataset` to `define_dataset`.  I expect this will be updated by work on DC-757 and DC-758.

Reply:
This comment is no longer valid.  `get_or_create_dataset` no longer exists.